storagemarket/provider.go

```diff
@@ -264,22 +297,37 @@ func (p *Provider) SubscribeNewDeals() (event.Subscription, error) {
 
 // SubscribeNewDeals subscribes to updates to a deal
 func (p *Provider) SubscribeDealUpdates(dealUuid uuid.UUID) (event.Subscription, error) {
-	dh, err := p.dealHandlers.get(dealUuid)
-	if err != nil {
-		return nil, err
+	dh := p.getDealHandler(dealUuid)
+	if dh == nil {
+		return nil, ErrDealHandlerFound
 	}
+
 	return dh.subscribeUpdates()
 }
 
-// CancelDeal cancels a deal and any associated data transfer
-func (p *Provider) CancelDeal(ctx context.Context, dealUuid uuid.UUID) error {
-	dh, err := p.dealHandlers.get(dealUuid)
-	if err != nil {
-		if xerrors.Is(err, ErrDealHandlerFound) {
-			return nil
-		}
-		return err
+// CancelDeal cancels the data-tarnsfer associated with a deal.
+func (p *Provider) CancelDealTransfer(ctx context.Context, dealUuid uuid.UUID) error {
+	dh := p.getDealHandler(dealUuid)
+	if dh == nil {
+		return ErrDealHandlerFound
```

Comment:
I'd suggest that we don't return an error when trying to cancel a data transfer for a deal that is not found; just return without doing anything. At this point the user probably doesn't care if it's no longer there, they just want to cancel it.